client/ayon_nuke/api/plugin.py

```diff
@@ -309,6 +311,26 @@ def set_selected_nodes(self, pre_create_data):
         else:
             self.selected_node = None
 
+    def update_instances(self, update_list):
+        for created_inst, changes in update_list:
+            instance_node = created_inst.transient_data["node"]
+
+            # update instance node name if product name changed
+            if "folderPath" or "task" in changes.changed_keys:
+                update_write_node_filepath(created_inst, changes)
```

Comment:
No need to check - the function itself already checks, right?
Also with this check here it wouldn't update the name on product name changes.
```suggestion
            update_write_node_filepath(created_inst, changes)
```

Reply:
Or maybe it makes more sense to take the `productName` updating of that `name` knob on the instance node out of that `update_write_node_filepath` function... so that function DOES actually ONLY update the filepath, nothing else.